lib/rucio/tests/test_replica_sorting.py

```diff
@@ -0,0 +1,195 @@
+        add_protocol(self.rse1, {'scheme': 'davs',
+                                 'hostname': 'davs.aperture.com',
+                                 'port': 443,
+                                 'prefix': '/test/chamber/',
+                                 'impl': 'rucio.rse.protocols.gfal.Default',
+                                 'domains': {
+                                     'lan': {'read': 2, 'write': 2, 'delete': 2},
+                                     'wan': {'read': 2, 'write': 2, 'delete': 2}}})
+        add_protocol(self.rse1, {'scheme': 'gsiftp',
+                                 'hostname': 'gsiftp.aperture.com',
+                                 'port': 8446,
+                                 'prefix': '/test/chamber/',
+                                 'impl': 'rucio.rse.protocols.gfal.Default',
+                                 'domains': {
+                                     'lan': {'read': 0, 'write': 0, 'delete': 0},
+                                     'wan': {'read': 3, 'write': 3, 'delete': 3}}})
+
+        add_protocol(self.rse2, {'scheme': 'gsiftp',
+                                 'hostname': 'gsiftp.blackmesa.com',
+                                 'port': 8446,
+                                 'prefix': '/lambda/complex/',
+                                 'impl': 'rucio.rse.protocols.gfal.Default',
+                                 'domains': {
+                                     'lan': {'read': 2, 'write': 2, 'delete': 2},
+                                     'wan': {'read': 1, 'write': 1, 'delete': 1}}})
+        add_protocol(self.rse2, {'scheme': 'davs',
+                                 'hostname': 'davs.blackmesa.com',
+                                 'port': 443,
+                                 'prefix': '/lambda/complex/',
+                                 'impl': 'rucio.rse.protocols.gfal.Default',
+                                 'domains': {
+                                     'lan': {'read': 0, 'write': 0, 'delete': 0},
+                                     'wan': {'read': 2, 'write': 2, 'delete': 2}}})
+        add_protocol(self.rse2, {'scheme': 'root',
+                                 'hostname': 'root.blackmesa.com',
+                                 'port': 1409,
+                                 'prefix': '//lambda/complex/',
+                                 'impl': 'rucio.rse.protocols.xrootd.Default',
+                                 'domains': {
+                                     'lan': {'read': 1, 'write': 1, 'delete': 1},
+                                     'wan': {'read': 3, 'write': 3, 'delete': 3}}})
+
+        replicas = [r for r in self.rc.list_replicas(dids=[{'scope': 'mock',
+                                                            'name': f['name'],
+                                                            'type': 'FILE'} for f in self.files],
+                                                     schemes=['root', 'gsiftp', 'davs'],
```

Comment:
What happens when schemes=None or schemes=[]?

Reply:
you will only get the highest ranking protocol per domain